lib/pythontarget.c

```diff
@@ -412,11 +412,8 @@ void destroy_action_capsule(PyObject* capsule) {
  * Individual ports can then later be accessed in Python code as port[idx].
  */
 PyObject* convert_C_port_to_py(void* port, int width) {
-    generic_port_instance_struct* cport;
-    if (width == -2) {
-        // Not a multiport
-        cport = (generic_port_instance_struct *)port;
-    }
+    generic_port_instance_struct* cport = (generic_port_instance_struct *)port;
+
```

Comment:
I wonder if the intended design was to keep this as-is, and to wrap line 433 (`FEDERATED_ASSIGN_FIELDS`) in `if (width == -2)`. Just guessing since it seems like `cport` is only being used when the port is not a multiport.

In fact it might have been clearer to move all of the code that touches `cport` into the branch at line 435-448.

Reply:
I agree with Peter here.  The proposed cast doesn't look correct for a multiport, and, although it is probably harmless, it seems misleading.